lib/streamlit/runtime/connection_factory.py

```diff
@@ -282,9 +309,18 @@ def connection_factory(
 
     # At this point, connection_class should be of type Type[ConnectionClass].
     try:
-        return _create_connection(
+        conn = _create_connection(
             name, connection_class, max_entries=max_entries, ttl=ttl, **kwargs
         )
+        if isinstance(conn, SnowparkConnection):
+            conn = deprecate_obj_name(
+                conn,
+                "SnowparkConnection",
+                "SnowflakeConnection",
+                "2024-04-01",
```

Comment:
I just noticed that this deprecation date is different from the `st.experimental_connection` deprecation date.
is this date based on Snowflake Connection release dates?

Reply:
Whoops -- the later one is the correct one. Will fix the `st.experimental_connection` deprecation date to match